modules/nf-core/variancepartition/dream/templates/dream.R

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env Rscript
+
+# Load libraries
+library("variancePartition")
+library("edgeR")
+library("BiocParallel")
+library("limma")
+
+# Load auxiliary helping functions
+
+#' Parse out options from a string without recourse to optparse
+#'
+#' @param x Long-form argument list like --opt1 val1 --opt2 val2
+#'
+#' @return named list of options and values similar to optparse
+parse_args <- function(x) {
+    args_list <- unlist(strsplit(x, ' ?--')[[1]])[-1]
+    args_vals <- lapply(args_list, function(x) scan(text = x, what = 'character', quiet = TRUE))
+    args_vals <- lapply(args_vals, function(z) { length(z) <- 2; z })
+    parsed_args <- structure(lapply(args_vals, function(x) x[2]), names = lapply(args_vals, function(x) x[1]))
+    parsed_args[ !is.na(parsed_args) ]
+}
+
+#' Flexibly read CSV or TSV files
+#'
+#' @param file Input file
+#' @param header Passed to read.delim()
+#' @param row.names Passed to read.delim()
+#'
+#' @return output Data frame
+#'
+read_delim_flexible <- function(file, header = TRUE, row.names = NULL, check.names = TRUE, stringsAsFactors = FALSE) {
+    ext <- tools::file_ext(basename(file))
+    if (ext == "tsv" || ext == "txt") {
+        separator <- "\\t"
+    } else if (ext == "csv") {
+        separator <- ","
+    } else {
+        stop(paste("Unknown separator for", ext))
+    }
+    cat("Reading file", basename(file), "with", ext, "separator\n")
+    df <- read.delim( file, sep = separator, header = header, row.names = row.names,
+                    check.names = check.names, stringsAsFactors = stringsAsFactors)
+    return(df)
+}
+
+
+# Options list
+opt <- list(
+    output_prefix      = "$meta.contrast_id",       # Prefix for output files
+    count_file         = "$counts",                 # File containing raw counts
+    sample_file        = "$samplesheet",            # File containing sample information
+    contrast_variable  = "$meta.contrast_variable", # Variable for contrast (e.g., "treatment")
+    contrast_reference    = "$meta.contrast_reference",# Reference level for the contrast
```

Comment:
Guidelines stipulate that custom meta fields should be avoided

https://nf-co.re/docs/guidelines/components/modules#types-of-meta-fields

Mandatory inputs (even non-file ones) should actually come via channels, like you see in deseq2 https://github.com/nf-core/modules/blob/f27081c5c554e34b4bcf9a51216f5aa98ba39028/modules/nf-core/deseq2/differential/main.nf#L11